test/test_examples.py

```diff
@@ -258,9 +258,7 @@ def many_missing(a, b, c):
     @pytest.mark.asyncio
     async def test_caching_with_batch(self):
         def trim_words(words, lens):
-            trimmed_words = []
-            for w, l in zip(words, lens):
-                trimmed_words.append(w[:l])
```

Comment:
Was this caught by `ruff` or did you change manually @akx?

Reply:
ruff was complaining about `l` being an ambiguous name, so while renaming, I fixed these up to the equivalent listcomp.